sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/DeadletterQueueSample.java

```diff
@@ -0,0 +1,213 @@
+     */
+    @Test
+    public void run() throws InterruptedException {
+        ServiceBusSenderAsyncClient senderAsyncClient = new ServiceBusClientBuilder()
+            .connectionString(connectionString)
+            .sender()
+            .queueName(queueName)
+            .buildAsyncClient();
+
+        // max delivery-count scenario
+        sendMessagesAsync(senderAsyncClient, 1);
+        exceedMaxDelivery(connectionString, queueName);
+
+        // fix-up scenario
+        sendMessagesAsync(senderAsyncClient, Integer.MAX_VALUE);
+        this.receiveMessagesAsync(connectionString, queueName);
+        this.PickUpAndFixDeadletters(connectionString, queueName, senderAsyncClient);
+
+        senderAsyncClient.close();
+    }
+
+    /**
+     * Send {@link ServiceBusMessage messages} to an Azure Service Bus Queue.
+     *
+     * @Param senderAsyncClient Service Bus Sender Async Client
+     * @Param maxMessages Maximum Number Of Messages
+     */
+    void sendMessagesAsync(ServiceBusSenderAsyncClient senderAsyncClient, int maxMessages) {
+        List<HashMap<String, String>> data =
+            GSON.fromJson(
+                "[" +
+                    "{'name' = 'Einstein', 'firstName' = 'Albert'}," +
+                    "{'name' = 'Heisenberg', 'firstName' = 'Werner'}," +
+                    "{'name' = 'Curie', 'firstName' = 'Marie'}," +
+                    "{'name' = 'Hawking', 'firstName' = 'Steven'}," +
+                    "{'name' = 'Newton', 'firstName' = 'Isaac'}," +
+                    "{'name' = 'Bohr', 'firstName' = 'Niels'}," +
+                    "{'name' = 'Faraday', 'firstName' = 'Michael'}," +
+                    "{'name' = 'Galilei', 'firstName' = 'Galileo'}," +
+                    "{'name' = 'Kepler', 'firstName' = 'Johannes'}," +
+                    "{'name' = 'Kopernikus', 'firstName' = 'Nikolaus'}" +
+                    "]",
+                new TypeToken<List<HashMap<String, String>>>() {
+                }.getType());
+
+        for (int i = 0; i < Math.min(data.size(), maxMessages); i++) {
+            final String messageId = Integer.toString(i);
+            ServiceBusMessage message = new ServiceBusMessage(GSON.toJson(data.get(i), Map.class).getBytes(UTF_8));
+            message.setContentType("application/json");
+            message.setSubject(i % 2 == 0 ? "Scientist" : "Physicist");
+            message.setMessageId(messageId);
+            message.setTimeToLive(Duration.ofMinutes(2));
+            System.out.printf("Message sending: Id = %s\n", message.getMessageId());
+            senderAsyncClient.sendMessage(message)
+                .doOnSuccess(onSuccess -> System.out.printf("\tMessage acknowledged: Id = %s\n", message.getMessageId()))
+                .block();
+        }
+    }
+
+    /**
+     * Receive {@link ServiceBusMessage messages} and dead letter it within an Azure Service Bus Queue
```

Comment:
Explain a bit about deadlettering when a message is received multiple times. A doc link is also helpful.
Just in case some users don't understand why "abandon" messages result in dead letter.

Reply:
According to your comment, fixed in new version.